brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/spec/Alien4CloudEntitySpecFactory.java

```diff
@@ -47,10 +52,11 @@ public Alien4CloudEntitySpecFactory(ManagementContext mgmt, ToscaFacade<Alien4Cl
             spec = (EntitySpec<?>) mgmt.getCatalog().createSpec(catalogItem);
 
         } else if (isComputeType(nodeId, toscaApplication)) {
-            spec = hasMultipleChildren
-                   ? EntitySpec.create(SameServerEntity.class)
-                   : EntitySpec.create(BasicApplication.class);
-
+            spec = EntitySpec.create(SameServerEntity.class);
+            spec.enricher(Enrichers.builder().transforming(Attributes.ADDRESS)
```

Comment:
Shouldn't this go on every spec? If yes, I think there's a case for a `ToscaAttributeMapping` spec modifier.